auth_manager.go

```diff
@@ -120,25 +184,32 @@ func (b *DefaultSessionManager) UpdateSession(keyName string, session *user.Sess
 		return nil
 	}
 
-	v, _ := json.Marshal(session)
-
-	if hashed {
-		keyName = b.store.GetKeyPrefix() + keyName
-	}
-
 	// async update and return if needed
 	if b.asyncWrites {
+		if atomic.LoadUint32(&b.shouldStop) > 0 {
+			return nil
+		}
 		b.renewSessionState(keyName, session)
 
-		if hashed {
-			go b.store.SetRawKey(keyName, string(v), resetTTLTo)
-			return nil
+		sessionMeta := &SessionUpdate{
+			isHashed: hashed,
```

Comment:
we could eliminate this field if we pass just keyName (we know here if prefix needs to be added)